### ISSUE TITLE
Cannot import eth-rlp due to `ModuleNotFoundError` for `typing_extensions`

### ISSUE THREAD

**Issue by aiman** [ethereum/eth-rlp#17]
### What happened?

After installing `eth-rlp`, it should be possible to import the package without error.

### Code that produced the error

```python
import eth_rlp
```

### Full error output

```shell
$ python -c 'import eth_rlp'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/test/.local/share/virtualenvs/test-eth-rlp-otJx7d21/lib/python3.12/site-packages/eth_rlp/__init__.py", line 5, in <module>
    from .main import (
  File "/home/test/.local/share/virtualenvs/test-eth-rlp-otJx7d21/lib/python3.12/site-packages/eth_rlp/main.py", line 18, in <module>
    from typing_extensions import Self  # move to typing when >= py311
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'typing_extensions'
```

### Fill this section in if you know how this could or should be fixed

The dependency on `typing_extensions` was added in commit da773689a83f567abd1b02b39431802af51993be, but there is no corresponding dependency in `setup.py`. This leads to import error on all currently supported Python versions `3.9`-`3.12`.

### eth-rlp Version

1.0.0

### Operating System

Linux

### Output from `pip freeze`

```shell
cytoolz==0.12.2
eth-hash==0.6.0
eth-rlp==1.0.0
eth-typing==4.0.0
eth-utils==3.0.0
hexbytes==0.3.1
rlp==4.0.0
toolz==0.12.0
```

**Comment by crjameson** [comment]
Had the same issue and installing typing extensions fixed it, so thank you for pointing that out :) 

Something like this might be a solution? as typing_extensions should'nt be neccesary for Python >3.11

```
if sys.version_info >= (3, 11):
    from typing import Self
else:
    from typing_extensions import Self
```

**Comment by BoboTiG** [comment]
Fixed with #19.
